client/components/Scenario/index.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Grid, Card } from 'semantic-ui-react';
+
+import SlideList from '@components/SlideList';
+import { setScenario, setSlides } from '@client/actions';
+
+class Scenario extends Component {
+    constructor(props) {
+        super(props);
+
+        // Check for data from route or props
+        this.state = Object.assign(
+            {},
+            { scenarioId: this.props.scenarioId },
+            this.props.location ? this.props.location.state : null,
+            this.props.match ? this.props.match.params : null
+        );
+
+        this.getScenarioData = this.getScenarioData.bind(this);
+        this.getScenarioSlides = this.getScenarioSlides.bind(this);
+
+        // Get data that hasn't been passed by the router
+        this.getScenarioData();
+        this.getScenarioSlides();
+    }
+
+    async getScenarioData() {
+        if (this.state.title && this.state.description) {
```

Comment:
Cool, this looks like it's handling local state if it exists, otherwise get the info from the backend. Kind of local vs remote?

Reply:
You got it! Since sometimes people would be clicking from the scenario list, I figured we might as well pass the title and description to this component if we have them. And then I'm using those as values to save in our Redux store if they're available.